code/datums/components/surgery_initiator.dm

```diff
@@ -0,0 +1,278 @@
+			continue
+		if(surgery.lying_required && target.body_position != LYING_DOWN)
+			continue
+		if(!surgery.self_operable && target == user)
+			continue
+		if(!surgery.can_start(user, target))
+			continue
+		for(var/path in surgery.target_mobtypes)
+			if(istype(target, path))
+				available_surgeries += surgery
+				break
+
+	return available_surgeries
+
+/// Does the surgery de-initiation.
+/datum/component/surgery_initiator/proc/attempt_cancel_surgery(datum/surgery/the_surgery, mob/living/patient, mob/user)
+	var/selected_zone = user.zone_selected
+	/// We haven't even started yet. Any surgery can be cancelled at this point.
+	if(the_surgery.status == 1)
+		patient.surgeries -= the_surgery
+		user.visible_message(
+			"<span class='notice'>[user] draws [parent] away from [patient]'s [parse_zone(selected_zone)].</span>",
+			"<span class='notice'>You remove [parent] from [patient]'s [parse_zone(selected_zone)].</span>",
+		)
+
+		qdel(the_surgery)
+		return
+
+	if(!the_surgery.can_cancel)
+		return
+
+	// Don't make a forced surgery implement cancel a surgery.
+	if(istype(the_surgery, forced_surgery))
+		return
+
+	var/obj/item/close_tool
+	var/obj/item/other_hand = user.get_inactive_hand()
+
+	var/is_robotic = !the_surgery.requires_organic_bodypart
+	var/datum/surgery_step/close_carbon = new /datum/surgery_step/generic/cauterize/premature()
+	var/datum/surgery_step/close_robo = new /datum/surgery_step/robotics/external/close_hatch/premature()
+	var/datum/surgery_step/chosen_close_step
+
+	if(is_robotic)
+		chosen_close_step = close_robo
+	else
+		chosen_close_step = close_carbon
+
+	if(isrobot(user))
+		if(!is_robotic)
+			// borgs need to be able to finish surgeries with just the laser scalpel, no special checks here.
+			close_tool = parent
+		else
+			close_tool = locate(/obj/item/crowbar) in user.get_all_slots()
+			if(!close_tool)
+				to_chat(user, "<span class='warning'>You need a prying tool in an inactive slot to stop the surgery!</span>")
+				return
+
+	else if(other_hand)
+		for(var/key in chosen_close_step.allowed_tools)
+			if(ispath(key) && istype(other_hand, key) || other_hand.tool_behaviour == key)
+				close_tool = other_hand
+				break
+
+	if(!close_tool)
+		to_chat(user, "<span class='warning'>You need a [is_robotic ? "prying": "cauterizing"] tool in your inactive hand to stop the surgery!</span>")
+		return
+
+	if(chosen_close_step.try_op(user, patient, selected_zone, close_tool, the_surgery))
+		// logging in case people wonder why they're cut up inside
+		log_attack(user, patient, "Prematurely finished \a [the_surgery] surgery.")
+		patient.surgeries -= the_surgery
+		qdel(the_surgery)
+
+/datum/component/surgery_initiator/proc/can_start_surgery(mob/user, mob/living/target)
+	if (!user.Adjacent(target))
+		return FALSE
+
+	// The item was moved somewhere else
+	if (!(parent in user))
+		to_chat(user, "<span class='warning'>How are you supposed to start an operation if you aren't holding the tool anymore?</span>")
```

Comment:
```suggestion
		to_chat(user, "<span class='warning'>You cannot start an operation if you aren't holding the tool anymore.</span>")
```
although it's a good questions, user feedback should probably be more straight forward and not rhetorical.